### DISCUSSION TITLE
Can we please figure out how to do this github pypi packaging thing better and explain it

### DISCUSSION THREAD

**Discussion by Fasti-au** [microsoft/autogen#3697] [Q&A] [answered]
https://github.com/microsoft/autogen/

FORKED FROM MICROSOFT
https://github.com/autogen-ai/autogen

both have current updated releases but the fork got autogen and the MS one got autogen-agentchat or something they called appropriate, yet no one seems to explain how we are meant to use this as a end user.......whats the right version to work with and why is there two that look like they are both active and claiming to be "autogen"

The versions don't make sense, and the one that isn't Microsoft got the original autogenic named package to give up its name.

please put info on your readme.MD s so we can figure out what to do .

**Comment by jackgerrits** [comment, marked as answer]
Hi @Fasti-au, hopefully we can provide some clarity on these topics and address some of the confusion with the info below. This has  been added also as [FAQs to the readme](https://github.com/microsoft/autogen/blob/main/README.md#faqs).

### Project Maintenance: 

We want to reaffirm our commitment to supporting both the original version of AutoGen (0.2) and the redesign (0.4).  AutoGen 0.4 is still work-in-progress, and we shared the code now to build with the community. There are no plans to deprecate the original AutoGen anytime soon, and both versions will be actively maintained.
 
### Forks: 
https://github.com/microsoft/autogen/ remains the only official repo for development and support of AutoGen. 
We are aware that there are thousands of forks of AutoGen, including many for personal development and startups building with or on top of the library. We are not involved with any of these forks and are not aware of any plans related to them. 

### Project Open-Source Status and License:

Our project remains fully open-source and accessible to everyone. We understand that some forks use different licenses to align with different interests. We will continue to use the most permissive license (MIT) for the project. 

### Packages:

Currently, we are unable to make releases to the `pyautogen` via PyPi package due to a change to package ownership that was done without our involvement. Additionally, we are moving to using multiple packages to align with the new design. Please see details [here](https://microsoft.github.io/autogen/dev/packages/index.html). 

Therefore, for the official package please use `autogen-agentchat~=0.2`.

### Communications and Discussions: 

We are unable to use Discord for project discussions. Therefore, we request that all discussions take place on https://github.com/microsoft/autogen/discussions/ going forward.  

### 0.4  is still evolving!
We are still actively developing AutoGen 0.4. One exciting new feature is the emergence of new SDKs for .NET. The python SDKs are further ahead at this time but our goal is to achieve parity. We aim to add additional languages in future releases. 

We are grateful to all the contributors to AutoGen 0.2 and we look forward to continuing to collaborate with everyone in the AutoGen community.

\- _Microsoft/AutoGen Team_

**Comment by afourney** [comment]
#3707 is merged to <s>staging</s> main. Here is the direct link https://github.com/microsoft/autogen/blob/main/README.md#faqs

**Comment by fahadshery** [comment]
thank you for the clarification but it's still a bit confusing:
1. PyPi shows autogen-agentchat 0.2.36
2. Also shows autogen 0.3.1
3. Also has pyautogen 0.3.1

Which one to use?

What is happening to the pyautogen? Is it still from Microsoft?
Are there any plans for `Docker` images?

**Reply by gagb** [reply]
@fahadshery as mentioned in @jackgerrits's response on behalf of the team, please use the `autogen-agentchat 0.2.36`. As also mentioned above [microsoft/autogen](https://github.com/microsoft/autogen/) remains the only official repo for development and support of AutoGen under MIT license.

**Comment by harish-coeff** [comment]
Looks like an internal power struggle, hope there is not much impact on users, people might be still using the fork (`pyautogen` or `autogen`), risky. If you lose control of a package name then you should make it clear so that users know that they are not on the official repo anymore. There might be many users who are using pyautogen, and just upgrading to the latest versions without knowing they are on a fork.
